### PR TITLE
Always create hive target directory if not present

### PR DESCRIPTION
**Pull request by rschlussel**
https://github.com/prestodb/presto/pull/16456 broke creation of empty
unpartitioned bucketed tables if they weren't using a temporary staging
directory, as the target directory would not get created.

Test plan - Unit test

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by NikhilCollooru — APPROVED** [review]
LGTM

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java [hunk 1]
```diff
@@ -1808,7 +1804,7 @@ else if (deleteEmptyDirectories && !recursiveDeleteResult.isDirectoryNoLongerExi
      * <p>
      * This method will not delete anything that's neither a directory nor a file.
      *
-     * @param queryIds prefix or suffix of files that should be deleted
+     * @param queryIds               prefix or suffix of files that should be deleted
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java:1807]: accidental change?

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java [hunk 2]
```diff
@@ -263,6 +264,11 @@ else if (table.getTableType().equals(EXTERNAL_TABLE)) {
             throw new PrestoException(NOT_SUPPORTED, "Table type not supported: " + table.getTableType());
         }
 
+        if (!table.getTableType().equals(VIRTUAL_VIEW)) {
+            File directory = new File(new Path(table.getStorage().getLocation()).toUri());
+            checkArgument(directory.exists(), "Table directory does not exist: %s", directory);
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java:269]: maybe `&& directory.isDirectory()` as well?

**Review by highker — APPROVED** [review]

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java (hunk 1 above)
Comment (rschlussel) [reply presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java:1807]: was automatically reformatted by intellij.  figured it was better to commit the autoformatting changes than fight with it all the time.

**Pull request merged**